packages/flutter/lib/src/foundation/key.dart

```diff
@@ -47,6 +49,23 @@ abstract class LocalKey extends Key {
   const LocalKey() : super.empty();
 }
 
+/// A key that is only equal to itself.
+///
+/// This cannot be created with a const constructor because that implies that
+/// all instantiated keys would be the same instance and therefore not be unique.
+class UniqueKey extends LocalKey {
```

Comment:
Moving this here makes sense to me, isn't foundation where the other Key classes are?  Though maybe someone with more context on this stuff than me should confirm.  @goderbauer?

Reply:
Moving this here seems fine.